core/src/main/java/io/grpc/ServiceDescriptor.java

```diff
@@ -31,53 +31,40 @@
 
 package io.grpc;
 
+import com.google.common.base.Preconditions;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
 /**
- * Definition of a method exposed by a {@link Server}.
- *
- * @see ServerServiceDefinition
+ * Descriptor for a service.
  */
 @ExperimentalApi("https://github.com/grpc/grpc-java/issues/1774")
-public final class ServerMethodDefinition<ReqT, RespT> {
-  private final MethodDescriptor<ReqT, RespT> method;
-  private final ServerCallHandler<ReqT, RespT> handler;
+public class ServiceDescriptor {
 
-  private ServerMethodDefinition(MethodDescriptor<ReqT, RespT> method,
-      ServerCallHandler<ReqT, RespT> handler) {
-    this.method = method;
-    this.handler = handler;
-  }
+  private final String name;
+  private final List<MethodDescriptor<?, ?>> methods;
 
-  /**
-   * Create a new instance.
-   *
-   * @param method the {@link MethodDescriptor} for this method.
-   * @param handler to dispatch calls to.
-   * @return a new instance.
-   */
-  public static <ReqT, RespT> ServerMethodDefinition<ReqT, RespT> create(
-      MethodDescriptor<ReqT, RespT> method,
-      ServerCallHandler<ReqT, RespT> handler) {
-    return new ServerMethodDefinition<ReqT, RespT>(method, handler);
+  public ServiceDescriptor(String name, MethodDescriptor<?, ?>... methods) {
```

Comment:
To sum up discussion offline: we could, but that would still be an addition to the class. We've got options, and when we need something we'll add another constructor parameter or use `with*()`; that's sounding fine to us.